### PR TITLE
PLT-5933 Fix drawer transition lag

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
When selecting a channel from the drawer the same drawer had a small lag when it was at 95% closed or something like that, with this PR we manually handle the interactions so we know when the drawer is fully closed

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5933tes

#### Device Information
This PR was tested on: 
* IOS 10.2.1 iPhone 6s
* Android 6.0.1 Samsung J5

### CODE REVIEW

File: app/actions/views/channel.js [hunk 1]
```diff
@@ -176,8 +176,8 @@ export function handleSelectChannel(channelId) {
         const {currentTeamId} = getState().entities.teams;
 
         await updateStorage(currentTeamId, {currentChannelId: channelId});
```
Comment (hmhealey) [inline app/actions/views/channel.js]: Would it help the lag here if we did these in parallel?

Comment (enahum) [inline app/actions/views/channel.js]: I can do that too

**Comment by enahum** [comment]
@hmhealey done

File: app/navigation/router.js [hunk 2]
```diff
@@ -216,6 +217,26 @@ class Router extends React.Component {
         };
     };
 
+    handleLeftDrawerClose = () => {
+        setTimeout(() => {
```
Comment (hmhealey) [inline app/navigation/router.js]: If I'm understanding these correctly, this prevents everything in `runAfterInteractions` blocks from running between the `createInteractionHandle` and `clearInteractionHandle` calls right? Why do we need to run the clear in a `setTimeout`?

File: app/navigation/router.js [hunk 3]
```diff
@@ -248,18 +269,22 @@ class Router extends React.Component {
...
                     openDrawerOffset={42}
                     onRequestClose={this.props.actions.closeDrawers}
-                    panOpenMask={0.3}
+                    panOpenMask={0.2}
                     panCloseMask={42}
                     panThreshold={0.2}
                     acceptPan={navigationProps.allowMenuSwipe}
                     negotiatePan={true}
-                    useInteractionManager={true}
+                    useInteractionManager={false}
```
Comment (hmhealey) [inline app/navigation/router.js]: Also, why do we not just let the drawer handle the `InteractionManager` itself?

**Pull request merged**